src/sdltiles.cpp

```diff
@@ -255,6 +259,45 @@ void init_interface()
 //***********************************
 //Non-curses, Window functions      *
 //***********************************
+void generate_alt_rect_texture()
+{
+#if SDL_BYTEORDER == SDL_BIG_ENDIAN
+    static const Uint32 rmask = 0xff000000;
+    static const Uint32 gmask = 0x00ff0000;
+    static const Uint32 bmask = 0x0000ff00;
+    static const Uint32 amask = 0x000000ff;
+#else
+    static const Uint32 rmask = 0x000000ff;
+    static const Uint32 gmask = 0x0000ff00;
+    static const Uint32 bmask = 0x00ff0000;
+    static const Uint32 amask = 0xff000000;
+#endif
+    SDL_Surface_Ptr altsurf( SDL_CreateRGBSurface( 0, 1, 1, 32, rmask, gmask, bmask, amask ) );
+    SDL_FillRect( altsurf.get(), NULL, SDL_MapRGB( altsurf->format, 255, 255, 255 ) );
+    SDL_Texture_Ptr alternate_rect_tex( SDL_CreateTextureFromSurface( renderer.get(), altsurf.get() ) );
+
+    // test to make sure color modulation is supported by renderer
+    if( SDL_SetTextureColorMod( alternate_rect_tex.get(), 0, 0, 0 ) ){
+        alternate_rect_tex_enabled = false;
+        SDL_DestroyTexture( alternate_rect_tex.get() );
```

Comment:
This will cause a double delete when `alternate_rect_tex` is destructed. *Don't* call any SDL destroy/free/delete function directly, call the `unique_ptr`s `reset()` function instead. This also means you can remove setting it to `NULL` - `reset` already does it.